crates/router/src/core/payments/payment_methods.rs

```diff
@@ -0,0 +1,225 @@
+            key_manager_state,
+            &payment_id,
+            &key_store,
+            merchant_account.storage_scheme,
+        )
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::PaymentNotFound)?;
+
+    validate_payment_status(payment_intent.status)?;
+
+    let client_secret = header_payload
+        .client_secret
+        .as_ref()
+        .get_required_value("client_secret header")
+        .change_context(errors::ApiErrorResponse::MissingRequiredField {
+            field_name: "client_secret header",
+        })?;
+
+    payment_intent.validate_client_secret(client_secret)?;
+
+    let payment_connector_accounts = db
+        .list_enabled_connector_accounts_by_profile_id(
+            key_manager_state,
+            profile.get_id(),
+            &key_store,
+            common_enums::ConnectorType::PaymentProcessor,
+        )
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("error when fetching merchant connector accounts")?;
+
+    let response =
+        hyperswitch_domain_models::merchant_connector_account::FlattenedPaymentMethodsEnabled::from_payment_connectors_list(payment_connector_accounts)
+            .perform_filtering()
+            .get_required_fields(RequiredFieldsInput::new())
+            .perform_surcharge_calculation()
+            .generate_response();
+
+    Ok(hyperswitch_domain_models::api::ApplicationResponse::Json(
+        response,
+    ))
+}
+
+/// Container for the inputs required for the required fields
+struct RequiredFieldsInput {}
+
+impl RequiredFieldsInput {
+    fn new() -> Self {
+        Self {}
+    }
+}
+
+/// Container for the filtered payment methods
+struct FilteredPaymentMethodsEnabled {
+    payment_methods_enabled: Vec<
+        hyperswitch_domain_models::merchant_connector_account::PaymentMethodsEnabledForConnector,
+    >,
+}
+
+impl FilteredPaymentMethodsEnabled {
+    fn get_required_fields(
+        self,
+        _input: RequiredFieldsInput,
+    ) -> PaymentMethodTypesEnabledWithRequiredFieldsContainer {
+        let required_fields_info = self
+            .payment_methods_enabled
+            .into_iter()
+            .map(
+                |payment_methods_enabled| PaymentMethodTypesEnabledWithRequiredFieldsElement {
+                    required_field: None,
+                    payment_method_type: payment_methods_enabled.payment_method,
+                    payment_method_subtype: payment_methods_enabled
+                        .payment_methods_enabled
+                        .payment_method_subtype,
+                },
+            )
+            .collect();
+
+        PaymentMethodTypesEnabledWithRequiredFieldsContainer {
+            payment_methods_enabled: required_fields_info,
+        }
+    }
+}
+
+/// Element container to hold the filtered payment methods with required fields
+struct PaymentMethodTypesEnabledWithRequiredFieldsElement {
```

Comment:
Avoid introducing Element And Container convention, Already we started using plurals for many and singular for single. Maintaining multiple convention is hard and not standard.